index.js

```diff
@@ -213,12 +219,20 @@ async function main() {
 
   app.post('/api/register', async (request, response) => {
     const { username } = request.body
+    const reValidUsername = /[a-zA-Z0-9_-]+/g
     let { password } = request.body
 
     if (!username || !password) {
       return
     }
 
+    if (!reValidUsername.test(username)) {
+      response.end(JSON.stringify({
+        error: 'username invalid'
+      }))
+      return
+    }
+
```

Comment:
Cool. I'll nitpick a couple things - this should have a status code (probably 400, Bad Request), and there should be a line break before `return`. (I know I'm being nitpicky, and that the changes I'm requesting are ones I made to my own code probably after you made this PR - sorry :) )

Reply:
>the changes I'm requesting are ones I made to my own code probably after you made this PR - sorry

haha yes